esi_leap/objects/contract.py

```diff
@@ -112,9 +113,16 @@ def fulfill(self, context=None):
         self.save(context)
 
     def expire(self, context=None):
+
+        if self.status == statuses.EXPIRED:
+            raise Exception("Already expired " + self.uuid)
+
         # unassign resource
         o = Offer.get_by_uuid(self.offer_uuid, context)
 
+        if o.status != statuses.AVAILABLE:
+            raise Exception("Offer not available " + o.uuid)
+
```

Comment:
I know you've brought up the reason for this before, but could you explain the need for this check again?

Reply:
My logic was that if an offer was not available, then it should have expired all of its contracts already.

Reply:
I'd remove this check, then; if the contract isn't expired, but the offer somehow has gotten into an expired state, I would think that we would want to expire that contract as fast as possible!